### PR TITLE
Update golangci-lint to latest v1.62.2

### PR DESCRIPTION
**Pull request by mstoykov**
## What?

update golangci-lint to latest version 

Quite a lot of gosec ones about integers under/overflows - I tried to fix the ones that weren't problematic, but there were quite a lot.

## Why?

Keeping up to date makes it easier to update and it is harder for IC to keep the exact version we use locally and unfortuantely there are always differences.

## Checklist

- [ ] I have performed a self-review of my code.
- [ ] I have added tests for my changes.
- [ ] I have run linter locally (`make lint`) and all checks pass.
- [ ] I have run tests locally (`make tests`) and all tests pass.
- [ ] I have commented on my code, particularly in hard-to-understand areas.

### CODE REVIEW

File: lib/executor/ramping_arrival_rate_test.go [hunk 1]
```diff
@@ -578,8 +578,8 @@ func sqrtRat(x *big.Rat) *big.Rat {
 	var z, a, b big.Rat
 	var ns, ds big.Int
 	ni, di := x.Num(), x.Denom()
-	z.SetFrac(ns.Rsh(ni, uint(ni.BitLen())/2), ds.Rsh(di, uint(di.BitLen())/2))
-	for i := 10; i > 0; i-- { // TODO: better termination
+	z.SetFrac(ns.Rsh(ni, uint(ni.BitLen())/2), ds.Rsh(di, uint(di.BitLen())/2)) //nolint:gosec
+	for i := 10; i > 0; i-- {                                                   // TODO: better termination
```
Comment (olegbespalov) [inline lib/executor/ramping_arrival_rate_test.go:582]: Why indent is so weird? :thinking:

**Review by olegbespalov — APPROVED** [review]
Keeping in mind number of the type conversions, `int`/`uint`/etc with omitting of the `gosec` it sounds that it might be so that we probably need some refactoring, on the other side it's for sure out of scope of this PR, so LGTM :+1:

File: lib/executor/ramping_arrival_rate_test.go (hunk 1 above)
Comment (mstoykov) [reply lib/executor/ramping_arrival_rate_test.go:582]: Do you want me to change this @olegbespalov ?

Comment (olegbespalov) [reply lib/executor/ramping_arrival_rate_test.go:582]: Whatever you prefer, I don't mind having this even if it looks weird.

Comment (mstoykov) [reply lib/executor/ramping_arrival_rate_test.go:582]: I will leave it for now then :sweat_smile:

**Review by ankur22 — APPROVED** [review]
LGTM 🚀

**Pull request merged**